include/mrdox/Plugin.hpp

```diff
@@ -12,16 +12,36 @@
 #define MRDOX_PLUGIN_HPP
 
 #include <mrdox/Platform.hpp>
-#include <mrdox/Generator.hpp>
+#include <mrdox/Support/Error.hpp>
+
+#include <string_view>
 
 namespace clang {
 namespace mrdox {
 
+class Generators;
+class Config;
+
+struct Plugin
+{
+    // MUST always come first
+    const std::string_view version;
+    Plugin(const std::string_view version) : version(version) {}
+
+    virtual
+    Error
+    registerGenerators(Generators & gen)
```

Comment:
The public API is designed so that there can be multiple invocations of each generator, each with a different Config and Corpus. For example to generate both single-page and multipage output in one run of the tool. The `Config` is tied to the `Corpus` and provided in the `Generator::build` member function call. At the time the plugin is loaded and initialized, there is no config yet.